app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Class Bolt_Boltpay_Block_Catalog_Product_Boltpay
+ *
+ * This block is used in boltpay/catalog/product/configure_checkout.phtml and boltpay/catalog/product/button.phtml templates
+ *
+ * This is defined in boltpay.xml config file
+ *
+ * The purpose is to add the Bolt to the Product Page,
+ * create the order on Bolt side through the javascript BoltCheckout.configureProductCheckout process.
+ *
+ */
+class Bolt_Boltpay_Block_Catalog_Product_Boltpay extends Mage_Core_Block_Template
+{
+    const CHECKOUT_TYPE_MULTI_PAGE  = 'multi-page';
+
+    /**
+     * Initiates sets up BoltCheckout config.
+     *
+     * @return string
+     */
+    public function getCartDataJsForProductPage()
+    {
+        try {
+            $currency = Mage::app()->getStore()->getCurrentCurrencyCode();
+
+            $productCheckoutCartItem = [];
+            $totalAmount = (float) 0;
+
+            /** @var Mage_Catalog_Model_Product $_product */
+            $_product = Mage::registry('current_product');
+            if (!$_product) {
+                $msg = 'Bolt: Cannot find product info';
+                Mage::helper('boltpay/bugsnag')->notifyException($msg);
+                return '""';
+            }
+
+            if ($_product->isInStock()) {
+                $productCheckoutCartItem[] = [
+                    'reference' => $_product->getId(),
+                    'price' => $_product->getPrice(),
+                    'quantity' => 1,
+                    'image' => $_product->getImageUrl(),
+                    'name' => $_product->getName(),
+                ];
+                $totalAmount = $_product->getPrice();
+            }
+
+            $productCheckoutCart = [
+                'currency' => $currency,
+                'items' => $productCheckoutCartItem,
+                'total' => $totalAmount,
+            ];
+
+             return json_encode($productCheckoutCart);
+        } catch (Exception $e) {
+            Mage::helper('boltpay/bugsnag')->notifyException($e);
+            return '""';
+        }
+    }
+
+    /**
+     * Collect callbacks for BoltCheckout.configureProductCheckout
+     *
+     * @param string $checkoutType
+     * @return string
+     */
+    public function getBoltCallbacks($checkoutType = self::CHECKOUT_TYPE_MULTI_PAGE)
```

Comment:
can we extract existing code in Boltpay.php into method and use it from here?

Reply:
done.